bowler/tool.py

```diff
@@ -148,6 +147,9 @@ def processed_file(
             if hunk:
                 hunks.append([a, b, *hunk])
 
+            features = _detect_future_features(new_text)
+            if "print_function" in features:
+                self.driver.grammar = pygram.python_grammar_no_print_statement
```

Comment:
I'm pretty sure this change will affect the following files as well.  If you look at the implementation of `refactor_string` in lib2to3 (which this is modeled after), it does a try/finally to restore it.

Reply:
ah right, I'll have a look and add this

Reply:
done in ff8c0d4273be7f35141d04e407509717f53c34e1

Reply:
Thanks.